core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/internal/CDOBasedRepository.java

```diff
@@ -348,5 +363,81 @@ public void handleCommitInfo(CDOCommitInfo commitInfo) {
 			notification.publish(events());
         }
 	}
-	
+
+	@Override
+	public boolean isConsistent(IBranchPath branch) {
+		List<CDOCommitInfo> cdoCommitInfos = getCDOCommitInfos(branch);
+		CommitInfos indexCommitInfos = getIndexCommitInfos(branch);
+		
+		boolean emptyDb = cdoCommitInfos.isEmpty();
+		boolean emptyIndex = indexCommitInfos.getItems().isEmpty();
+
+		if (emptyDb && emptyIndex) {
+			return true; // empty dataset
+		}
+		
+		if (emptyDb ^ emptyIndex) {
+			LOG.error("{} is in inconsistent state. CDO {} but INDEX {}", getCdoRepository().getRepositoryName(), contentMessage(emptyDb), contentMessage(emptyIndex));
+			return false; // either CDO or index was deleted but not the other.
+		}
+		
+		return !hasInvalidCDOTimeStamps(cdoCommitInfos, getLast(indexCommitInfos).getTimeStamp(), branch);
```

Comment:
during import, for versions: there is only one commit for the index, but there are multiple commits for CDO, thus the size never matches. For this reason @cmark suggested to check for only after index CDO commits